tmt/beakerlib.py

```diff
@@ -219,30 +221,71 @@ def fetch(self) -> None:
             # Prepare path, clone the repository, checkout ref
             assert self.parent.workdir
             directory = os.path.join(self.parent.workdir, self.dest, self.repo)
+            clone_dir = os.path.join(directory, 'clone')
             # Clone repo with disabled prompt to ignore missing/private repos
             try:
                 if self.url:
                     # Shallow clone to speed up testing and
                     # minimize data transfers if ref is not provided
-                    tmt.utils.git_clone(self.url, directory, self.parent,
+                    tmt.utils.git_clone(self.url, clone_dir, self.parent,
                                         env={"GIT_ASKPASS": "echo"}, shallow=self.ref is None)
+
+                    # Detect the default branch from the origin
+                    try:
+                        self.default_branch = tmt.utils.default_branch(clone_dir)
+                    except OSError:
+                        raise tmt.utils.GeneralError(
+                            f"Unable to detect default branch for '{clone_dir}'. "
+                            f"Is the git repository '{self.url}' empty?")
+                    # Use the default branch if no ref provided
+                    if self.ref is None:
+                        self.ref = self.default_branch
+                    # Check out the requested branch
+                    try:
+                        self.parent.run(Command('git', 'checkout', self.ref), cwd=clone_dir)
+                    except tmt.utils.RunError:
+                        # Fallback to install during the prepare step if in rpm format
+                        if self.format == 'rpm':
+                            self.parent.debug(f"Invalid reference '{self.ref}'.")
+                            raise LibraryError
+                        self.parent.fail(
+                            f"Reference '{self.ref}' for library '{self}' not found.")
+                        raise
+
+                    # Copy only the required library
+                    assert self.path is not None
+                    library_path: str = os.path.join(clone_dir, self.path.strip('/'))
+                    local_library_path: str = os.path.join(directory, os.path.basename(self.path))
+                    if not os.path.exists(library_path):
+                        self.parent.debug(f"Failed to find library {self} at {self.url}")
+                        shutil.rmtree(clone_dir, ignore_errors=True)
+                        raise LibraryError
+
+                    tmt.utils.copytree(library_path, local_library_path, dirs_exist_ok=True)
```

Comment:
This misses any .fmf files used for inheritance (e.g. main.fmf in git root with common metada as contact, require common for all libraries...). For example
```
/var/tmp/tmt/CURRENT/plans/default/discover/default-0/libs
└── repo1
    ├── foo
    │   ├── lib.sh
    │   └── main.fmf
    └── main.fmf
```
but this PR will change it to (`repo1/main.fmf` is missing)
```
/var/tmp/tmt/NEW/plans/default/discover/default-0/libs
└── repo1
    └── foo
        ├── lib.sh
        └── main.fmf
```

Reply:
Now all inherited metadata should be in one file, thanks to d0a27c2c88ba3d54e1c69dcec01f8e330e7a402a